feature/experimental/bgp/ate_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,467 @@
+	nbrPath := statePath.Neighbor(ateAttrs.IPv4)
+
+	// Configure BGP Neighbor on the DUT
+	t.Logf("Start DUT Network Instance and BGP Config")
+	configureNetworkInstance(t)
+	// Clear any existing config
+	fptest.LogYgot(t, "DUT BGP Config before", dutConfPath, dutConfPath.Get(t))
+	dutConfPath.Replace(t, nil)
+	dutConf := bgpCreateNbr(dutAS, ateAS, 0, "", "", holdTime0, ConnTime0, NegHoldTime0)
+	dutConfPath.Replace(t, dutConf)
+
+	t.Logf("configure port and BGP configs on ATE")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+	iDut1 := topo.AddInterface(ateAttrs.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateAttrs.IPv4CIDR()).WithDefaultGateway(dutAttrs.IPv4)
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dutAttrs.IPv4).WithLocalASN(ateAS).WithTypeExternal()
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	t.Logf("Verify BGP session state : ESTABLISHED")
+	nbrPath.SessionState().Await(t, time.Second*100, telemetry.Bgp_Neighbor_SessionState_ESTABLISHED)
+
+	//TO DO: WAITING FOR THE FIX FROM ATE TO SEND NOTIFICATION TO DUT TO VERIFY ERROR CODE ON DUT
+
+	/*t.Logf("Send Notification from ATE ")
+	// ADD CODE HERE TO SEND NOTIFICATION FROM ATE
+
+	t.Logf("Verify BGP session state : ACTIVE")
+	nbrPath.SessionState().Await(t, time.Second*100, telemetry.Bgp_Neighbor_SessionState_ACTIVE)
+
+	t.Logf("Verify Error code received on DUT: BgpTypes_BGP_ERROR_CODE_CEASE")
+	code := nbrPath.Messages().Received().LastNotificationErrorCode().Get(t)
+	if code != telemetry.BgpTypes_BGP_ERROR_CODE_CEASE {
+		t.Errorf("On disconnect: expected error code %v, got %v", telemetry.BgpTypes_BGP_ERROR_CODE_CEASE, code)
+	}*/
+
+	//clean config on DUT and ATE
+	topo.StopProtocols(t)
+	dutConfPath.Replace(t, nil)
+}
+
+// TestParameters is to verify BGP connection parameters like timers, router-id, md5 authentication.
+func TestParameters(t *testing.T) {
+	ateIP := ateAttrs.IPv4
+	dutIP := dutAttrs.IPv4
+	dut := ondatra.DUT(t, "dut")
+
+	//Configure network instance with network-instance type and router-id
+	configureNetworkInstance(t)
+
+	dutConfPath := dut.Config().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	statePath := dut.Telemetry().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateIP)
+
+	//configure loopback to use it for explicit router id
+	configureDUTLoop(t, dut, dutlo0Attrs1)
+	loopIPAddr := getLoopIP(t)
+
+	cases := []struct {
+		name      string
+		dutConf   *telemetry.NetworkInstance_Protocol_Bgp
+		ateConf   *ondatra.ATETopology
+		wantState *telemetry.NetworkInstance_Protocol_Bgp
+		skipMsg   string
+	}{
+		{
+			name:    "basic internal",
+			dutConf: bgpCreateNbr(dutAS, dutAS, 0, "", "", holdTime0, ConnTime0, NegHoldTime0),
```

Comment:
Both bgpCreateNbr and configureATE have a long list of very similar arguments, and it's not apparent by reading the code here what the arguments are. It's possible someone might have swapped the order of some arguments without realizing it.

Consider creating a shared type like this:

```
type bgpParams struct {
  localAS int
  peerAS int
  nbrLocalAS int
  peerIP string
  authPwd string
  ...
}

func (params *bgpParams) forDUT() *telemetry.NetworkInstance_Protocol_Bgp {
  // This is the new bgpCreateNbr
}

func (params *bgpParams) forATE() *ondatra.ATETopology {
  // This is the new configureATE
}
```

Here in the test case, we can just have dutParams and ateParams, which we then convert to the OC struct and ATE topology in the body.